### PR TITLE
use EVP interface

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Use OpenSSL's EVP interface instead of the deprecated low-level APIs
Enterprise companion PR: https://github.com/arangodb/enterprise/pull/398

From https://www.openssl.org/docs/man1.1.0/man3/SHA256_Final.html:
> Applications should use the higher level functions EVP_DigestInit(3) etc. instead of calling the hash functions directly.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *scripts/unittest catch*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8690/

### CODE REVIEW

File: lib/Basics/files.cpp [hunk 1]
```diff
@@ -2666,3 +2667,39 @@ bool TRI_GETENV(char const* which, std::string& value) {
...
+    EVP_MD_CTX_free(_context);
+#else
+    EVP_MD_CTX_destroy(_context);
+#endif
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, "unable to initialize SHA256 processor");
+  }
+}
+
+TRI_SHA256Functor::~TRI_SHA256Functor() {
+  EVP_MD_CTX_free(_context);
```
Comment (ObiWahn) [inline lib/Basics/files.cpp]: No version switch here?

Comment (jsteemann) [reply lib/Basics/files.cpp]: Good catch... Fixed!

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8695/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8700/

**Pull request merged**